chore(ci): CI scripts cleanup, add Node 14 runs, bump checkout action

Simek opened the pull request:
## Motivation

This PR clean up the GitHub action workflow files and adds Node 14 run to the "Yarn v1 E2E" run and Node 15 to the "Yarn v2 E2E" run to ensure that not only the minimal version of Node, but also `Active LTS` and `Current` do no produce any errors for the users.

I have also bumped all the remaining `actions/checkout@v1` to `v2` - some of workflows were already using `v2`. Also the Node version declaration has been updated to the recommended syntax - https://github.com/actions/setup-node.

Additionally I have moved the Yarn version set command at the beginning of script so initial installation would also be performed using Yarn v2.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Need to run the CI.

## Related PRs

No.

slorber commented on .github/workflows/e2e-test.yml:
```diff
@@ -56,7 +56,6 @@ jobs:
       - name: Setup test-website project for Yarn v2
         run: |
           cd test-website
-          yarn set version berry
```
Was wondering why it's removed? 
Are we sure to use Yarn2 in this flow without that line?

slorber replied:
ah ok it's been moved :)

Simek replied:
It has been moved a few step earlier, as stated in the summary to ensure that `install` is also performed by v2.

Simek commented:
Changing Node from `15` to `14` resolved the Yarn v2 E2E workflow issue, it look like this PR brought to table two problems:
* Yarn v2 do not work with the Docusaurus workspace so contributors using it by default would struggle - but it's nice that you can switch that fast), it looks like types are required now, I have added bunch of them, but  TS compiler also become very strict for some reason and do not tolerate unknow and any anymore
* `init` do not work correctly on Node 15, which is shipped with NPM 7 by default, this might be due to:
  * changed `npx` implementation - https://blog.npmjs.org/post/626173315965468672/npm-v7-series-beta-release-and-semver-major
  * other NPM 7 changes

slorber commented:
Thanks, seems to be fine for now, but we'll try to figure out these issues you reported
